### PR TITLE
server: less failures

### PR DESCRIPTION
**Pull request by tharvik**
* even an invalid tx should be logged in the accounts (but not changing the balance)
* fix "already existing" in recent txs
* add a timeout in processing the txs, help when users do send invalid tx

### CODE REVIEW

File: src/bin/server/accounts/account.rs [hunk 1]
```diff
@@ -7,8 +7,8 @@ pub enum Error {
     Underflow,
 }
 
-/// Contains the balance for a user
-#[derive(Debug)]
+/// Contains the balance and last processed sequence for a user
```
Comment (ineiti) [inline src/bin/server/accounts/account.rs]:
```suggestion
/// Contains the balance and the latest processed sequence for a user
```

We hope there is another one...

File: src/bin/server/accounts/mod.rs [hunk 2]
```diff
@@ -217,58 +217,85 @@ impl AccountsHandler {
...
-            .expect("to get final sequence");
+        let (final_balance, final_sequence) =
+            get_balance_and_sequence(&accounts, user_pubkey).await;
 
         assert_eq!(initial_balance, final_balance);
         assert!(initial_sequence < final_sequence);
     }
+
+    #[tokio::test]
+    async fn transfer_too_much_fails_and_increase_sequence() {
```
Comment (ineiti) [inline src/bin/server/accounts/mod.rs]:
```suggestion
    async fn transfer_too_much_fails_and_increases_sequence() {
```

**Review by ineiti — APPROVED** [review]
General question: it seems to me that there is quite some logic now with regard to how transactions are handled. Is this all described in the paper?

**Comment by tharvik** [comment]
> it seems to me that there is quite some logic now with regard to how transactions are handled. Is this all described in the paper?

Nope, there isn't any description on how to handle coins or accounts in the paper. It's only describing the communication mechanisms that can help to do so.

**Comment by ineiti** [comment]
OK - another one of these "just implement it"...

**Pull request merged**